src/main/java/dev/hephaestus/glowcase/block/HyperlinkBlock.java

```diff
@@ -29,38 +29,37 @@ public VoxelShape getOutlineShape(BlockState state, BlockView world, BlockPos po
 		return OUTLINE;
 	}
 
+	@Nullable
 	@Override
-	public void onPlaced(World world, BlockPos pos, BlockState state, LivingEntity placer, ItemStack itemStack) {
-		if (!world.isClient) {
-			if (placer instanceof ServerPlayerEntity player && player.isCreative() && world.canPlayerModifyAt(player, pos)) {
-				HyperlinkChannel.openScreen(player, pos);
+	public BlockEntity createBlockEntity(BlockPos pos, BlockState state) {
+		return new HyperlinkBlockEntity(pos, state);
+	}
+
+	@Override
+	public void onPlaced(World world, BlockPos pos, BlockState state, LivingEntity placer, ItemStack stack) {
+		if (world.isClient && placer instanceof PlayerEntity player && canEditGlowcase(player, pos)) {
+			//load any ctrl-picked NBT clientside
+			NbtCompound blockEntityTag = BlockItem.getBlockEntityNbt(stack);
+			if(blockEntityTag != null && world.getBlockEntity(pos) instanceof HyperlinkBlockEntity be) {
+				be.readNbt(blockEntityTag);
 			}
```

Comment:
Previously Glowcase used a server-initiated packet to open the editing screen, which also happened to ensure the client got the block entity NBT before the editing screen attempted to open. Now that the GUI opening is initiated by the client immediately when they place the block, if there is a `BlockEntityTag` I need to load it clientside before opening the screen. (For some reason the vanilla `BlockEntityTag` stuff only happens serverside.)